File: 0607.py
import sys
N, M = map(int, input().split())
a = [0] * N
for l in sys.stdin.readlines():
    a = [n+m for n, m in zip(a, map(int, l.split()))]+[0]
    i, j, total = 0, 1, a[0]
    while j <= N:
        if total == 777:
            print("YES")
            exit()
        if total < 777:
            total += a[j]
            j += 1
        else:
            total -= a[i]
            i += 1
print("NO")

File: 0607.kt

fun main(args: Array<String>) { 
  val (ws, n) = readLine()!!.split(" ").map { it.toInt() }
  val xs = (1..ws).map { 0 }.toMutableList()
  
  var check = false
  val arrs = (1..n).map { 
    val arr = readLine()!!.split(" ").map { it.toInt() }
    arr
  }
  
  total@for( arr in arrs ) {
    arr.mapIndexed { index, x ->
      xs[index] += x
    }
    // 尺取サーチ
    for( left in (0..xs.size-1) ) {
      for( right in (left..xs.size-1) ) {
        if( xs.slice( (left..right) ).sum() == 777 ) {
          check = true
          break@total
        }
      }
    }
  }

  if(check) 
    println("YES")
  else
    println("NO")
}
